src/generic_ui/polymer/contact.html

```diff
@@ -57,119 +68,139 @@
-          </div>
-
-          <div hidden?='{{ contact.gettingConsentState != GettingConsentState.NO_OFFER_OR_REQUEST }}'>
-            <paper-button raised class='highlightedButton' on-tap='{{ request }}'>
-              Ask for access
-            </paper-button>
-          </div>
-
-        </div> <!-- end of getControls -->
-
-
-
-        <div id='shareControls' hidden?='{{globalSettings.mode!=uProxy.Mode.SHARE}}'>
-
-          <div hidden?='{{ contact.sharingConsentState != SharingConsentState.LOCAL_OFFERED_REMOTE_ACCEPTED }}'>
-            <p class='preButtonText'>You've given them access.</p>
-            <paper-button raised on-tap='{{ cancelOffer }}'>Revoke Access</paper-button>
-          </div>
-
-          <div hidden?='{{ contact.sharingConsentState != SharingConsentState.LOCAL_OFFERED_REMOTE_NO_ACTION }}'>
-            <paper-button raised on-tap='{{ cancelOffer }}'>Cancel Offer</paper-button>
-          </div>
-
-          <div hidden?='{{ contact.sharingConsentState != SharingConsentState.REMOTE_REQUESTED_LOCAL_NO_ACTION }}'>
-            <p class='preButtonText'>They requested access through you.</p>
-            <paper-button raised class='highlightedButton' on-tap='{{ offer }}'>Grant</paper-button>
-            <paper-button raised class='highlightedButton' on-tap='{{ ignoreRequest }}'>Ignore</paper-button>
-          </div>
-
-          <div hidden?='{{ contact.sharingConsentState != SharingConsentState.REMOTE_REQUESTED_LOCAL_IGNORED }}'>
-            <p class='preButtonText'>They requested access through you.</p>
-            <paper-button raised on-tap='{{ unignoreRequest }}'>Stop ignoring requests</paper-button>
-          </div>
-
-          <div hidden?='{{ contact.sharingConsentState != SharingConsentState.NO_OFFER_OR_REQUEST }}'>
-            <p class='preButtonText'>You have not granted them access.</p>
-            <paper-button raised class='highlightedButton' on-tap='{{ offer }}'>Offer Access</paper-button>
-          </div>
-
-        </div> <!-- end of shareControls -->
-
-      </div> <!-- end of #expandedControls -->
+    <div class='nameRow' on-click='{{ toggle }}' tabindex='-1'
+        class='frame {{ isOnline ? "online" : "offline" }}'
+        center horizontal layout>
+      <img class='avatar' src='{{ contact.imageData }}' />
+      <div flex class='name'>{{ contact.name }}</div>
+      <img src='../icons/getting_animated.gif' class='accessIcon'
+          hidden?='{{ globalSettings.mode!=uProxy.Mode.GET || !contact.isSharingWithMe }}'>
+      <img src='../icons/sharing_animated.gif' class='accessIcon'
+          hidden?='{{ globalSettings.mode!=uProxy.Mode.SHARE || !contact.isGettingFromMe }}'>
+      <core-icon icon="expand-more"
+        hidden?='{{ !(globalSettings.mode==uProxy.Mode.GET && !contact.getExpanded && !contact.isSharingWithMe) && !(globalSettings.mode==uProxy.Mode.SHARE && !contact.shareExpanded && !contact.isGettingFromMe) }}'>
```

Comment:
Can you comment on this hidden statement?  also maybe a TODO to investigate ways we can toggle item visibility with if/else-if/else type logic (so that we can be sure not to show both the getting/sharing icon along with the expand/collapse icon) - this was possible in Angular and may be possible in polymer